app/code/Magento/InventoryInStorePickup/Plugin/InventoryApi/SourceRepository/LoadInStorePickupOnGetPlugin.php

```diff
@@ -7,49 +7,41 @@
 
 namespace Magento\InventoryInStorePickup\Plugin\InventoryApi\SourceRepository;
 
-use Magento\Framework\Api\ExtensionAttributesFactory;
+use Magento\Framework\DataObject;
 use Magento\InventoryApi\Api\Data\SourceInterface;
 use Magento\InventoryApi\Api\SourceRepositoryInterface;
-use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+use Magento\InventoryInStorePickup\Model\Source\SetExtensionAttributes;
 
 class LoadInStorePickupOnGetPlugin
 {
     /**
-     * @var \Magento\Framework\Api\ExtensionAttributesFactory
+     * @var SetExtensionAttributes
      */
-    private $extensionAttributesFactory;
+    private $setExtensionAttributes;
 
     /**
-     * @param \Magento\Framework\Api\ExtensionAttributesFactory $extensionAttributesFactory
+     * @param SetExtensionAttributes $setExtensionAttributes
      */
-    public function __construct(ExtensionAttributesFactory $extensionAttributesFactory)
-    {
-        $this->extensionAttributesFactory = $extensionAttributesFactory;
+    public function __construct(
+        SetExtensionAttributes $setExtensionAttributes
+    ) {
+        $this->setExtensionAttributes = $setExtensionAttributes;
     }
 
     /**
      * Enrich the given Source Objects with the In-Store pickup attribute
      *
      * @param SourceRepositoryInterface $subject
-     * @param SourceInterface $source
+     * @param SourceInterface|DataObject $source
```

Comment:
That's cheating. We can't typehint for SourceInterface and rely on Dataobject in reality

Reply:
Fixed